crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -953,23 +945,13 @@ impl CheckpointStore {
         &self,
         transactions: &CheckpointContents,
     ) -> SuiResult<bool> {
-        let new_transactions = self
+        // TODO: What mechanisms are there to ensure these not-yet-executed transactions
+        // will eventually be executed?
```

Comment:
When it comes to making checkpoints the active service in `execution_driver`:
https://github.com/MystenLabs/sui/tree/main/crates/sui-core/src/authority_active/execution_driver
will execute any "extra" transactions that this validator does not have to allow for a signature.

In case the checkpoint was generated elsewhere, the active part of the checkpoint process should run the sync process to execute all transactions in the checkpoint. This is not done yet I believe as it depends on sync which we do not have.

Should we augment the sync issue / make a new issue relating to checkpoints pointing this out? I assumed it would come naturally with sync.